demo/src/main/java/io/weet/demo/controllers/UserProfileController.java

```diff
@@ -50,13 +52,15 @@ public String userProfile(Model model) {
     public String allergenDelete(@RequestParam(name = "name") String name) {
         if (allergenList.containsKey(name)) {
             allergenList.remove(name);
+            userService.deleteAllergen(user.getEmail(), name);
```

Comment:
you already have the user object. There is no need to perform another search by email in the userservice. Instead, pass the whole user object to these functions.

Reply:
Following [this](https://github.com/SiddharthDatar/Weet/pull/24/files#r762494732) comment, we could use that function here and pass in allergenList. It's a map so you're going to have to convert the Map --> ArrayList, which is done in the same file